src/map/packet_system.cpp

```diff
@@ -3206,26 +3206,27 @@ void SmallPacket0x04E(map_session_data_t* const PSession, CCharEntity* const PCh
                     return;
                 }
 
-                // 2. Remove item from inventory
+                // 2. Remove item from inventory. Cache name because this method may actually delete the item pointer
                 auto deductQuantity = -(int32)(quantity != 0 ? 1 : PItem->getStackSize());
                 auto itemId         = charutils::UpdateItem(PChar, LOC_INVENTORY, slot, deductQuantity);
+                auto itemName       = PItem->getName();
```

Comment:
Need to dereference the itemName before calling UpdateItem (since that's where the pointer potentially gets changed)

Reply:
ah. Item->getName() does not return a copy? that makes sense.

Reply:
oh im an idiot. I placed this AFTER the update call....

Reply:
Wll GetName() return a copy of the name?

Reply:
hmm I see, it just returns the field as std::string. So probably need to copy that?

Reply:
should be fine using auto.  You might have to do `itemName.c_str()' to cast it to a char* type later on, but I think it works this way.

Reply:
Nope, no c_str() required.  Just tested locally.  Just need to move the GetName function before the UpdateItem call and you're good.

Reply:
Even if its not required, will the auto create a copy for sure?

Reply:
Removed c_str()